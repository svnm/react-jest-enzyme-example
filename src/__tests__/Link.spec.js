import React from 'react'
import Link from '../Link'

/* setup test renderer */
import renderer from 'react-test-renderer'

/* setup enzyme */
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })
import {shallow} from 'enzyme'

/* setup jsdom */
var jsdom = require('jsdom')
const { JSDOM } = jsdom
const window = (new JSDOM('')).window
global.window = window
global.document = window.document

test('Link renders correctly and matches snapshot', () => {
  const component = renderer.create(
    <Link page="https://reactjs.org">React</Link>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="https://reactjs.org">React</Link>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Link renders the correct elements', () => {
  const wrapper = shallow(
    <Link page="https://reactjs.org">
      <p>React</p>
    </Link>
  );

  expect(wrapper.find('.normal').length).toEqual(1)
  expect(wrapper.find('p').length).toEqual(1)
  expect(wrapper.find('p').first().text()).toEqual('React')
});
