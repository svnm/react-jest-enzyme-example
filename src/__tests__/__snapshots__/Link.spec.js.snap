// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://reactjs.org"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  React
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://reactjs.org"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  React
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://reactjs.org"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  React
</a>
`;

exports[`Link renders correctly and matches snapshot 1`] = `
<a
  className="normal"
  href="https://reactjs.org"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  React
</a>
`;
